### PR TITLE
Add mapped domain enhanced menu

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* Add the enhanced domain navigation menu to mapped domain status screen
* The "Update your DNS record" card is using the same code to generate the destination as the "Change your name servers & DNS records" for registered domains
* Connect your domain is used only for mapped domains that support Domain Connect. Since I wrote the texts and chose the icon I would appreciate a copy review @fditrapani 
* The "Transfer your domain to WordPress.com" - I changed the secondary text since determining if a domain is locked is slow and might be unreliable
* Added tracking to "Transfer your domain to WordPress.com" so we can see if it's useful - `calypso_domain_management_mapped_transfer_click` event

It might look like different combinations of the following depending on the domain state and type:

Screenshot 2020-04-27 at 22 59 08
Screenshot 2020-04-27 at 23 00 05

#### Testing instructions

* Open a mapped domain and check the navigation menus. Try with mapped domains pointing/not pointing with us. Also domains that do not have our name servers. Also test with subdomains. Also check what happens when you click on the "Transfer your domain to WordPress.com"

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx [hunk 1]
```diff
@@ -246,8 +254,28 @@ class DomainManagementNavigationEnhanced extends React.Component {
...
+		if ( ! isEligibleForInboundTransfer ) {
+			return null;
+		}
+
+		return (
+			<DomainManagementNavigationItem
+				onClick={ this.handleTransferDomainClick }
+				path={ domainTransferIn( selectedSite.slug, domain.name, true ) }
+				materialIcon="sync_alt"
+				text={ translate( 'Transfer your domain to WordPress.com' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:271]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Transfer Domain to WordPress.com' )` *ES Score: 7*

**Comment by fditrapani** [comment]
Thanks for putting this together @hambai! I appreciate the extra care on the copy and icon! 

While I was testing this with a mapped domain I purchased, the status didn’t look quite right because it said the domain was active even though I had never set it up. I was expecting to see the status and messaging about setting up my mapped domain. This would also be the ideal place to put the domain connect button, unless there's a need for it after the domain is connected?

This is what I'm seeing:
Screen Shot 2020-04-27 at 10 16 07 PM

> Connect your domain is used only for mapped domains that support Domain Connect. Since I wrote the texts and chose the icon I would appreciate a copy review @fditrapani

I like the icon but as I mentioned, I think this option should appear in the messaging if a domain isn't setup because we won't need it after it's connected right? We can update this in another PR if you'd like to ship the other changes too.

> The "Transfer your domain to WordPress.com" - I changed the secondary text since determining if a domain is locked is slow and might be unreliable

Good call on changing this copy rather than leaving it to load for ever. Can we change it to `Manage your site and domain all in one place`?

> Added tracking to "Transfer your domain to WordPress.com" so we can see if it's useful - calypso_domain_management_mapped_transfer_click event

Thank you!

**Comment by hambai** [comment]
> While I was testing this with a mapped domain I purchased, the status didn’t look quite right because it said the domain was active even though I had never set it up. I was expecting to see the status and messaging about setting up my mapped domain. This would also be the ideal place to put the domain connect button, unless there's a need for it after the domain is connected?

Yep, I'd prefer to push this in a separate PR.

**Comment by fditrapani** [comment]
Sounds good! no problem.

**Comment by olaseni** [comment]
The link (i.e. navigation item) that says `Review your domain security` essentially navigates to the same destination as  `Find similar domains`, as can be seen [here](https://github.com/Automattic/wp-calypso/pull/41494/files#diff-d463f0d7c87f0280360121911a8bdf1dR400).

Is that intentional? This probably pre-dates this PR but it looks like something we should address.

Also when my domain mapping has external name servers, the text description says `Destination: external service`. It may not be obvious to users that the DNS records entered will not work until the name servers resolve. Is there a way to depict this? Perhaps wording like `Your records will/may not work yet`?

**Comment by hambai** [comment]
> Is that intentional? This probably pre-dates this PR but it looks like something we should address.

Yes, this will be addressed in a separate PR I'm working on right now. The enhanced navigation is feature flagged so this won't be visible to our users till we finish all the stuff.

> Also when my domain mapping has external name servers, the text description says `Destination: external service`. It may not be obvious to users that the DNS records entered will not work until the name servers resolve. Is there a way to depict this? Perhaps wording like `Your records will/may not work yet`?

Interesting... didn't it show "Your domain mapping has not been setup. You need to update your name servers, at the company you purchased your domain, to..." message in the domain status card? Can you share the domain mapping you tested with (in Slack)?

**Comment by olaseni** [comment]
> Interesting... didn't it show "Your domain mapping has not been setup. You need to update your name servers, at the company you purchased your domain, to..." message in the domain status card? Can you share the domain mapping you tested with (in Slack)?

It does show that indeed. However I'm trying to experience that screen as the user. 

The status card gives general information about the state of the domain, but some users may not connect that information to their ability to modify DNS records, since the navigation item/button is available to them.

**Comment by hambai** [comment]
> The status card gives general information about the state of the domain, but some users may not connect that information to their ability to modify DNS records, since the navigation item/button is available to them.

I understand. Some times you would actually want to be able to update the DNS records on our side before the name servers are pointed at us. However I agree that we might want to put a notice that the records might not be publicly visible since the name servers are still not pointing to us. I don't think that should be in the navigation item though - probably a warning in the DNS editor is a better place for this. I'll make a card to fix this

**Pull request merged**